src/query_router.rs

```diff
@@ -215,6 +216,50 @@ impl QueryRouter {
         // server it'll go to if the query parser is enabled.
         if matches.len() != 1 {
             debug!("Regular query, not a command");
+            if comment_shard_routing_enabled {
+                // comment_shard_routing_enabled is true, but no sharding comment was found
+                match self.pool_settings.no_shard_specified_behavior {
+                    Some(behavior) => {
+                        match behavior {
+                            NoShardSpecifiedHandling::Random => {
```

Comment:
You can match the option and the enum both in one go to reduce a level of indentation
```rust
match option {
  Some(NoShardSpecifiedHandling::Random) => {}
Some(NoShardSpecifiedHandling::RandomHealthy) => {}
...
```